dags/operators/moz_dataproc_operator.py

```diff
@@ -283,7 +283,10 @@ def _build_gce_cluster_config(self, cluster_data):
 
         # Fetch current nested dict and add nested keys
         cluster_config_new = cluster_data['config']
-        cluster_config_new.update({'softwareConfig' : {'optionalComponents': self.optional_components }})
+
+        software_config_new = cluster_config_new['softwareConfig']
```

Comment:
What if this key does not exist?

Reply:
It will exist, whole dictionary is created here: https://github.com/mozilla/telemetry-airflow/blob/d2257dcf52b8ca5f847b53463b4e96c94e3015b7/dags/operators/moz_dataproc_operator.py#L387

Granted, this isn't ideal but it's consistent with the rest of this file.